src/utils/parse.rs

```diff
@@ -83,3 +84,106 @@ where
+                    if token.try_push(input[position]).is_err() {
+                        return Err(ParseVectorError::TooLongNumber { position });
+                    }
+                }
+                b'/' => {
+                    token.reverse();
+                    let s = unsafe { std::str::from_utf8_unchecked(&token[..]) };
+                    dims = s
+                        .parse::<usize>()
+                        .map_err(|_| ParseVectorError::BadParsing { position })?;
+                }
+                b'}' => {
+                    token.clear();
+                    break 'a position;
+                }
+                b' ' => continue,
+                _ => return Err(ParseVectorError::BadCharacter { position }),
+            }
+        }
+        return Err(ParseVectorError::BadParentheses { character: '}' });
+    };
+    let mut indexes = Vec::<u32>::new();
+    let mut values = Vec::<T>::new();
+    let mut index: u32 = 0;
+    for position in left + 1..right {
+        let c = input[position];
+        match c {
+            b'0'..=b'9' | b'a'..=b'z' | b'A'..=b'Z' | b'.' | b'+' | b'-' => {
+                if token.is_empty() {
+                    token.push(b'$');
+                }
+                if token.try_push(c).is_err() {
+                    return Err(ParseVectorError::TooLongNumber { position });
+                }
+            }
+            b',' => {
+                if !token.is_empty() {
+                    // Safety: all bytes in `token` are ascii characters
+                    let s = unsafe { std::str::from_utf8_unchecked(&token[1..]) };
+                    let num = f(s).ok_or(ParseVectorError::BadParsing { position })?;
+                    indexes.push(index);
+                    values.push(num);
+                    token.clear();
+                } else {
+                    return Err(ParseVectorError::TooShortNumber { position });
+                }
+            }
+            b':' => {
+                if !token.is_empty() {
+                    // Safety: all bytes in `token` are ascii characters
+                    let s = unsafe { std::str::from_utf8_unchecked(&token[1..]) };
+                    index = s
+                        .parse::<u32>()
+                        .map_err(|_| ParseVectorError::BadParsing { position })?;
+                    token.clear();
+                } else {
+                    return Err(ParseVectorError::TooShortNumber { position });
+                }
+            }
+            b' ' => (),
+            _ => return Err(ParseVectorError::BadCharacter { position }),
+        }
+    }
+    if !token.is_empty() {
+        let position = right;
+        // Safety: all bytes in `token` are ascii characters
+        let s = unsafe { std::str::from_utf8_unchecked(&token[1..]) };
+        let num = f(s).ok_or(ParseVectorError::BadParsing { position })?;
+        indexes.push(index);
+        values.push(num);
+        token.clear();
+    }
```

Comment:
What if `input = '{0:}/5'`?

Reply:
Can we use json parsing directly?

Reply:
> Can we use json parsing directly?

Why mention json parsing?

Reply:
> > Can we use json parsing directly?
> 
> Why mention json parsing?

After stripped length, {1:5, 2:6, 3:7} or {} could be a valid json

Reply:
It just makes things even more complex and harder to reason correctness.